File: ProblemSolve/1863.cpp
/********************************
 * Sum of All Subset XOR Totals *
 ********************************/

class Solution {
public:
    void backtracking(vector<int>& nums, int idx, int curXor, int& ans){
        if(idx >= nums.size()){
            return;
        }
        
        for(int i=idx; i<nums.size(); ++i){
            ans += (curXor ^ nums[i]);
            backtracking(nums, i + 1, curXor ^ nums[i], ans);
        }
    }
    
    int subsetXORSum(vector<int>& nums) {
        int ans = 0;
        backtracking(nums, 0, 0, ans);
        
        return ans;
    }
};

File: ProblemSolve/1863.cs
/********************************
 * Sum of All Subset XOR Totals *
 ********************************/

public class Solution {
    int ans = 0;
    
    public void Backtracking(int[] nums, int idx, int curXor){
        if(idx >= nums.Length){
            return;
        }
        
        for(int i=idx; i<nums.Length; ++i){
            ans += (curXor ^ nums[i]);
            Backtracking(nums, i + 1, curXor ^ nums[i]);
        }
    }
    
    public int SubsetXORSum(int[] nums) {
        Backtracking(nums, 0, 0);
        
        return ans;
    }
}